magento/pwa-studio
# Increase test coverage for `packages/peregrine/lib/talons/CheckoutPage`

## Description

Increase coverage for files under `packages/peregrine/lib/talons/CheckoutPage`

**NOTE:**
_The uncovered lines in the report represent branching logic paths that appear to be unreachable._

Line 12 in `CheckoutPage/AddressBook/useAddressCard.js`: Address cannot be undefined because the destructuring in Line 5 would cause an error before it could reach Line 12.

Line 132 in `CheckoutPage/PaymentInformation/useCreditCard.js`: Cannot test the ( `false` || `true` ) path for this line since the only time `dropInLoading` is set to `false`, the `stepNumber` is always set to 0.

## Related Issue





Closes PWA-996

## Acceptance

any developer

### Verification Stakeholders

any developer

### Specification



### Verification Steps

1. Run tests for this specific directory: `yarn test --collectCoverageFrom="packages/peregrine/lib/talons/CheckoutPage/**/*.js" packages/peregrine/lib/talons/CheckoutPage/`
2. Verify acceptable test coverage

## Screenshots / Screen Captures (if appropriate)

## Checklist

* I have added tests to cover my changes, if necessary.
* I have added translations for new strings, if necessary.
* I have updated the documentation accordingly, if necessary.
